sktime_dl/utils/_models.py

```diff
@@ -25,3 +27,45 @@ def save_trained_model(model, model_save_directory, model_name, save_format='h5'
             file_name = model_name + '.hdf5'
         path = Path(model_save_directory) / file_name
         model.save(path)  # Add save_format here upon migration from keras to tf.keras
+
+
+def check_is_fitted(estimator, msg=None):
+    """Perform is_fitted validation for estimator.
+
+    Adapted from sklearn.utils.validation.check_is_fitted
+
+    Checks if the estimator is fitted by verifying the presence and positivity of
+    self.is_fitted_
+
+    Parameters
+    ----------
+    msg : string
+        The default error message is, "This %(name)s instance is not fitted
+        yet. Call 'fit' with appropriate arguments before using this
+        estimator."
+
+        For custom messages if "%(name)s" is present in the message string,
+        it is substituted for the estimator name.
+
+        Eg. : "Estimator, %(name)s, must be fitted before sparsifying".
+
+    Returns
+    -------
+    None
+
+    Raises
+    ------
+    NotFittedError
+        If the attributes are not found.
+    """
+    if isclass(estimator):
+        raise TypeError("{} is a class, not an instance.".format(estimator))
+    if msg is None:
+        msg = ("This %(name)s instance is not fitted yet. Call 'fit' with "
+               "appropriate arguments before using this estimator.")
+
+    if not hasattr(estimator, 'fit'):
+        raise TypeError("%s is not an estimator instance." % (estimator))
+
+    if not hasattr(estimator, 'is_fitted_') or not estimator.is_fitted_:
```

Comment:
I'd follow one convention only, so we don't need the trailing underscore anymore (to avoid confusion with sklearn's convention).

Reply:
Yeah that's reasonable